src/components/FilterSearch.tsx

```diff
@@ -49,8 +49,20 @@ export interface FilterSearchProps {
    * Defaults to "Search here...".
    */
   placeholder?: string,
-  /** Whether to trigger a search when an option is selected. Defaults to false. */
+  /**
+   * Whether to trigger a search when an option is selected. Defaults to false.
+   *
+   * @deprecated Use the `onSelect` prop instead.
+   */
   searchOnSelect?: boolean,
+  /** A function which is called when a filter is selected. */
+  onSelect?: (
+    currentFilter: StaticFilter | undefined,
```

Comment:
perhaps the params to onSelect can be an object instead? that way somebody can pick out just the params they need for what they want, without having to do have unused params like `onSelect(_unused1, _unused2, newFilter)`

Reply:
I thought about it, but I can't really think of a time when people wouldn't be using all of the params to perform some kind of filter search selecting behavior. I think it would be rare if that happened and even then I think they would use most of the params. Because of that, I didn't think it was worth making an object for the params because most people would then have to add code to either destructure the object or refer to the params as `params.blah`. what do you think?

Reply:
By "add code to destructure the object" do you mean add curly braces around the function like `onSelect({ currentFilter })` instead of `onSelect(currentFilter)`? I feel like that's not an issue and from my experience most people prefer destructuring things. It's also more flexible long term. I think the only downside is that it's more annoying for people to rename the variables if they don't like our names (they can just do `onSelect({ currentFilter: myFilter })`, though)

I was imagining the most common use for onSelect being somebody wanting to customize "what happens when a new filter is selected". They might not even care about FilterSearch maintaining it's own "currentFilter" state. This was something Alex and I talked about when product first started asking for FilterSearch changes - Alex thought somebody might NOT want FilterSearch to deselect the previous filters that were selected.
Anyways for that specific use case somebody would ONLY care about "newFilter", so they would have to do `onSelect(_currentFilter, _setCurrentFilter, newFilter)`. Which again is really not a big deal but I definitely prefer destructuring. We could also reorder the params ourselves in a way that we think would be used more commonly, and I think I would probably put newFilter at the very start, but I don't want to presume I can think of every use case and what the ideal ordering would be. Destructuring also makes it easier to add in additional params later if we need to.

Reply:
Got it. I forgot for a moment that they could just add the curly braces instead of needing `onSelect(params) { const { currentFilter } = params; ... }`. And destructuring would definitely be better in terms of flexibility and future scalability. I'll update to make an object!